package ESB;

message Command {
	enum Cmd {
		ERROR = 1;
		RESPONSE = 2;
		NODE_HELLO = 3;
		PING = 4;
		PONG = 5;
		INVOKE = 6;
		REGISTER_INVOKE = 7;
		REGISTER_INVOKE_OK = 8;
	};
	required Cmd cmd = 1;
	required string payload = 2;
	optional string identifier = 3;
	optional string guid_from = 4;
	optional string guid_to = 5;
	optional string source_proxy_guid = 6;
	optional string target_proxy_guid = 7;
	optional int32 version = 8;
	optional uint32 start_time = 9;
	optional int32 timeout_ms = 10;
}
